src/app/shared/services/latestSearchesUtil.js

```diff
@@ -0,0 +1,29 @@
+export default class LatestSearchesUtil {
+    constructor($window, latestSearches) {
+        this.$window = $window;
+        this.latestSearches = latestSearches;
+    }
+
+    getLatestSearches() {
+        return this.latestSearches = JSON.parse(this.$window.localStorage.getItem('latestSearches')) || [];
+    };
+
+    addCity(city) {
+        if (city == undefined || city == null || city.trim() === '' || this.latestSearches.includes(city)) {
+            return;
+        }
+
+        if (city === this.latestSearches[0]) {
```

Comment:
isn't this check redundant with the `this.latestSearches.includes(city)` of the previous 'if'?